src/controllers/scripting/legacy/scriptconnection.cpp

```diff
@@ -5,19 +5,36 @@
 
 void ScriptConnection::executeCallback(double value) const {
     Trace executeCallbackTrace("JS %1 callback", key.item);
-    const auto args = QJSValueList{
-            value,
-            key.group,
-            key.item,
-    };
-    QJSValue func = callback; // copy function because QJSValue::call is not const
-    QJSValue result = func.call(args);
-    if (result.isError()) {
-        if (controllerEngine != nullptr) {
-            controllerEngine->showScriptExceptionDialog(result);
+
+    if (!m_callbackExecuting && !m_preventRecursiveCalls) {
+        m_callbackExecuting = true;
+        m_preventRecursiveCalls = true;
+
```

Comment:
Setting that flag here leads to never having the critical warning. I think it has to be only checked to inhibit the warning below.